actions/tag/increment-tag/entrypoint

```diff
@@ -38,14 +41,19 @@ main() {
   fi
 
   if [ -n "${current_version}" ]; then
+    printf "Current version provided: %s\n" "${current_version} - using this as the tag"
     tag="${current_version}"
   else
     if [[ -z "$(git rev-list --tags --max-count=1)" ]]; then
-      echo "There are no previous tags"
+      printf "There are no previous tags - using v0.0.0 as the tag"
       previous="v0.0.0"
     else
       # List all tags of HEAD, sorted in ascending semver order; use the last one
-      previous="$(git tag --sort "version:refname" --points-at "$(git rev-list --tags --max-count=1)" | tail -n 1)"
+      all_previous="$(git tag --sort "version:refname" --merged)"
+      printf "Last 10 previous tags: %s\n" "$(tail -n 10 <<< "${previous}")"
+
+      previous="$(tail -n 1 <<< "${all_previous}")"
```

Comment:
Not sure I get the benefit of doing this - why do we care about getting the last 10? Or how is it helpful for debugging?

Reply:
yeah it was just helpful for debugging because the most recent tag wasn't quite enough info to figure out why it was picking the wrong tag.

Reply:
update - chatted with Rob about this - printing out 10 tags to help understand any weirdness in repo version history (as would have been useful in the failed stack build linked in the PR description)